libs/application/templates/general-fishing-license/src/graphql/queries.ts

```diff
@@ -37,3 +37,11 @@ export const queryFishingLicense = gql`
     }
   }
 `
+
+export const PAYMENT_STATUS = gql`
+  query status($applicationId: String!) {
```

Comment:
The name of the query is too generic and might override existing ones.

Additionally, always query the GraphQL ID of the type as well in order to update the store.

Reply:
Do you mean query the __typename?

Reply:
@darrikonn

Reply:
No (sorry for late answers, missed this somehow).
So now there's a query name called `status` in the GraphQL schema that's accessible for every single application. The name needs to relate to e.g. payment. Check https://docs.devland.is/technical-overview/api-design-guide/graphql-naming-conventions#integrating-naming-conventions-into-shared-api for reference.

Regarding the GraphQL ID. I'm assuming that you're querying the payment status of the application. You however don't query the application id as well, so Apollo doesn't know how to construct its cache, making the payment status not accessible in other components that might need it. A better approach would be to do something like:
```typescript
query application($input: ApplicationInput!) {
  application(input: $input) {
    id
    paymentStatus
  }
}
```
where variables are:
```typescript
variables: {
  input: {id: applicationId}
}
```

This way, the Apollo cache can construct the cache by setting the paymentStatus to the appropriate application because the GraphQL ID is fetched as well (id).

Reply:
I ended up removing this part since Arnór made a reusable PaymentPending ui component.